HighlightCorrectButton did not unselect button when navigating to unknown View

schuettecarsten opened the pull request:
This PR fixes a regression in my latest changes. HighlightCorrectButton did not unselect the latest Button if a View is shown that is not known in the HamburgerMenu.

nba0cschuette committed:
Fix HighlightCorrectButton did not unselect the latest Button if a view is shown that is not known in the HamburgerMenu

JerryNixon commented:
This `UpdateInternalBindingValues` is really a good addition. 
Can you refresh from master so I can merge this?

schuettecarsten commented:
Updated. Please note that the main idea in my changes is that the button is only touched when the selection really changes. So I moved some code into the ReferenceEquals-if-block. But you're right, it should not be a problem to update the internal binding values every time, but if you call it for the new button, you should also call it for the old button.

nba0cschuette committed:
Fix merge

nba0cschuette committed:
Fix merge/compile (again)

schuettecarsten commented:
My latest commit is neccessary to make this PR work after the changes in PR #1388. So please make sure to merge #1388 before this PR or revert commit 249f368.
